src/utils/utils.ts

```diff
@@ -1,13 +1,51 @@
 import featuredItemsData from "../data/featureditems.json";
+let featuredItemArray = [];
+
+/**
+ * Sets global featuredItemArray so other functions can use it.
+ **/
+export const setFeaturedItemArray = () => {
```

Comment:
You can update this function so that you don't have to make this logic check and call twice in different functions:
```
if (featuredItemArray.length === 0) {
  setFeaturedItemArray();
}
```.

Check if `featuredItemArray.length` is not 0 and if it is not, then return. Otherwise, go through the process of setting it.

```
if (featuredItemArray.length !== 0) {
  // The global array already has what it needs.
  return;
}

// If it's empty, now is the time to generate and update the array.
let itemsArray = [];
for (let i = 0; i < featuredItemsData.featuredItems.images.length; i++) {
  itemsArray.push(featuredItemsData.featuredItems.images[i].split(".")[0]);
}
// set featuredItemArray
featuredItemArray = itemsArray;

return featuredItemArray;
```

Now, in `generateRandomImageID` and `featuredImageID` (or anywhere else), can you call `setFeaturedItemArray()` directly without any if statement wrapper. All the logic will be inside of `setFeaturedItemArray`.